File: 0014.LongestCommonPrefix/LongestCommonPrefix.c
char *longestCommonPrefix(char **strs, int strsSize)
{
        if (!strsSize)
                return "";
        static char out[1024];
        int outlen;
        snprintf(out, 1024, "%s", strs[0]);
        outlen = strlen(out);
        int cnt = strsSize - 1;
        while (cnt)
        {
                if (strncmp(out, strs[cnt], outlen))
                {
                        char *a = out, *b = strs[cnt];
                        while (*a == *b)
                        {
                                a++;
                                b++;
                        }
                        *a = 0;
                        outlen = strlen(out);
                }
                cnt--;
        }
        return out;
}

File: 0014.LongestCommonPrefix/LongestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strslen = len(strs)
        if not strslen:
            return ''
        if strslen == 1:
            return strs[0]

        cnt = 0
        while True:
            xx = False
            for st in strs:
                try:
                    if st[cnt] != strs[0][cnt]:
                        xx = True
                        break
                except:
                    xx = True
            if xx == True:
                if cnt == 0:
                    cnt = -1
                break
            cnt += 1

        if cnt >= 0:
            return strs[0][:cnt]
        return ""
